src/Uno.Toolkit.UI/Controls/ZoomContentControl/ZoomContentControl.cs

```diff
@@ -0,0 +1,782 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Numerics;
+using Uno.Disposables;
+using Uno.Extensions.Specialized;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using System.Threading.Tasks;
+
+#if IS_WINUI
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Controls.Primitives;
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Input;
+using Microsoft.UI.Xaml.Media;
+using Microsoft.UI.Xaml.Navigation;
+using Microsoft.UI.Input;
+#else
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+using Windows.UI.Input;
+using Windows.Devices.Input;
+#endif
+
+
+namespace Uno.Toolkit.UI;
+
+[TemplatePart(Name = TemplateParts.RootGrid, Type = typeof(Grid))]
+[TemplatePart(Name = TemplateParts.Presenter, Type = typeof(ContentPresenter))]
+[TemplatePart(Name = TemplateParts.VerticalScrollBar, Type = typeof(ScrollBar))]
+[TemplatePart(Name = TemplateParts.HorizontalScrollBar, Type = typeof(ScrollBar))]
+public partial class ZoomContentControl : ContentControl
+{
+	private static class TemplateParts
+	{
+		public const string RootGrid = "PART_RootGrid";
+
+		public const string Presenter = "PART_Presenter";
+
+		public const string HorizontalScrollBar = "PART_ScrollH";
+		public const string VerticalScrollBar = "PART_ScrollV";
+	}
+
+	private Grid? _grid;
+	private ContentPresenter? _presenter;
+	private ScrollBar? _scrollV;
+	private ScrollBar? _scrollH;
+
+	private Point _lastPosition = new Point(0, 0);
+
+	private (bool Horizontal, bool Vertical) _movementDirection = (false, false);
+
+	private bool IsAllowedToWork => (IsEnabled && IsActive && _presenter is not null);
+
+	public bool ResetWhenNotActive { get; set; } = true;
+
+	public Size AvailableSize
```

Comment:
AvailableSize is typically refers to value passed in uwp measure. I would pick something else to avoid confusion.

Reply:
since, this is solely used in `ZoomToCanvas()`, consider just inlining this property:
```cs
var hZoom = (ActualWidth - AdditionalMargin.Left + AdditionalMargin.Right) / ViewPortWidth;
```